lib/sprout/src/main/kotlin/org/partiql/sprout/generator/target/kotlin/KotlinGenerator.kt

```diff
@@ -192,4 +188,51 @@ class KotlinGenerator(private val options: KotlinOptions) : Generator<KotlinResu
         builder.addFunction(copy.build())
         impl.addFunction(copyImpl.build())
     }
+
+    /**
+     * Adds `equals` method to the core abstract class
+     */
+    private fun KotlinNodeSpec.Product.addEqualsMethod() {
+        val equalsFunctionBodyBuilder = CodeBlock.builder().let { body ->
+            body.addStatement("if (this === other) return true")
+            body.addStatement("if (other !is %T) return false", this.clazz)
+            this.props.forEach { prop ->
+                body.addStatement("if (%N != other.%N) return false", prop.name, prop.name)
+            }
```

Comment:
I think we should deeply compare collections as well

Reply:
This should be deeply comparing as-is. I just pushed tests to show some examples. Thanks!